src/Orleans.Runtime/Hosting/EndpointOptions.cs

```diff
@@ -29,4 +39,65 @@ public class EndpointOptions
 
         //public bool BindToAny { get; set; }
     }
+
+    /// <summary>
+    /// Represent a network address with its mask
+    /// </summary>
+    public class NetworkAddress
+    {
+        /// <summary>
+        /// Address of the network
+        /// </summary>
+        public IPAddress Address { get; set; }
+
+        /// <summary>
+        /// Mask of the network
+        /// </summary>
+        public IPAddress Mask { get; set; }
+
+        public AddressFamily AddressFamily => Address.AddressFamily;
+
+        /// <summary>
+        /// Create a new NetworkAddress
+        /// </summary>
+        public NetworkAddress()
+            : this(IPAddress.Any, IPAddress.Any)
+        { }
+
+        /// <summary>
+        /// Create a new NetworkAddress
+        /// </summary>
+        public NetworkAddress(IPAddress address, IPAddress mask)
+        {
+            if (address.AddressFamily != mask.AddressFamily)
```

Comment:
This invariant can be broken by using the getters and setters. Maybe we should make the setters private

Reply:
I agree, but I was worried about deserialization of those fields. You don't think that can be an issue with private setter?